### ISSUE TITLE
Auto-selection of new transfers

### ISSUE THREAD

**Issue by thep50** [nicotine-plus/nicotine-plus#2901]
Nicotine+ version:  3.3.2
Operating System/Distribution: macOS Monterey 12.7.3 + Windows 11

I've noticed since version 3.3.0 a feature that was introduced on both Windows and macOS that's causing issues for me.

On both the Downloads and Uploads tabs, every time a new transfer is added or initiated, it gets auto-selected. I like this when adding new downloads so I can see instantly where the new downloads are in the list, but it becomes an issue when you have the context menu open for a certain selection and a new transfer is initiated: the selection changes to the new transfer while the context menu is still open, and even jumps the page up or down to the new transfer. This is especially an issue if it's a long transfer list.

Wondering if this could be adjusted in some way.

**Comment by slook** [comment]
It was not the intention to allow the selection to be altered whilst the context menu is open. There must have been some oversight.

Do you think it will be satisfactory enough to block the auto-selection feature whilst a context menu is currently open, or is something more than this required?

**Comment by thep50** [comment]
Well I suppose the auto selection is still an issue for me mainly in the uploads tab. It makes it hard to look through a long list if the selection keeps moving to the bottom.

**Comment by slook** [comment]
It needs to work similar to the chat text view, in that auto-scrolling will only occur when it is already scrolled to the bottom.

**Comment by mathiascode** [comment]
It's only supposed to auto-select new transfers if the tab isn't active.

**Comment by thep50** [comment]
Would this fix have included in the latest unstable version? I've installed 3.3.3.dev1 as of just now and it's still auto-selecting new transfers with the context menu open.

In any case, I realized it might be best to fully disable auto-select, at least on the upload tab. I've found sometimes when my uploads to one user are stuck and keep retrying in fast succession, every time it retries it auto-selects away from anything else I want to select, making it impossible to do anything else.
